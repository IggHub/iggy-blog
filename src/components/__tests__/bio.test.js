import React from "react"
import Bio from "../bio"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      avatar: {
        childImageSharp: {
          fixed: {},
        },
      },
      site: {
        siteMetadata: {
          authorAlias: "testAuthor",
          social: {
            twitter: "someTwitterHandle",
          },
        },
      },
    })
  )
})

describe("Bio", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Bio />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
