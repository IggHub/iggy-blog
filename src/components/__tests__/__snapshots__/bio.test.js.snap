// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginBottom": "4.375rem",
    }
  }
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "borderRadius": "100%",
        "display": "inline-block",
        "height": undefined,
        "marginBottom": 0,
        "marginRight": "0.875rem",
        "minWidth": 50,
        "overflow": "hidden",
        "position": "relative",
        "width": undefined,
      }
    }
  >
    <picture>
      <img
        alt="testAuthor"
        onLoad={[Function]}
        style={
          Object {
            "borderRadius": "50%",
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 0.5s",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><img src=\\"\\" alt=\\"testAuthor\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <p>
    The name's 
    <strong>
      testAuthor
    </strong>
    . I am a developer in Dallas. I write about useful things I found on internet.
     
    <a
      href="https://twitter.com/someTwitterHandle"
    >
      Everybody should totally follow him on twitter.
    </a>
  </p>
</div>
`;
